tcpproxy/tcpproxy.go

```diff
@@ -51,6 +51,7 @@ type testTCPProxy struct {
 	recvFg           failuregen.FailureGenerator
 	acceptFg         failuregen.FailureGenerator
 	stats            ProxyStats
+	sync.Mutex
```

Comment:
I think the change to combine `stats` and `mutex` caused some confusion. I was trying to get only two attributes (mutex and stats) grouped up in an anonymous struct. It was meant as documentation (to express we only care to protect `ProxyStats` not all the other attributes (which are never modified)).

So declaration for `stats` would then something like 
```
stats struct {
  sync.Mutex
  value ProxyStats
}
```
etc.

Reply:
Also, because we have locks now, let us change the three counters in `ProxyStats` to `int64` (no atomic).

It is confusing when atomics are used with lock.